extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/DefaultGroupsSignEncryptUnitTest.java

```diff
@@ -0,0 +1,58 @@
+package io.quarkus.jwt.test;
+
+import java.net.HttpURLConnection;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+import io.smallrye.jwt.algorithm.KeyEncryptionAlgorithm;
+import io.smallrye.jwt.build.Jwt;
+
+public class DefaultGroupsSignEncryptUnitTest {
+    private static Class<?>[] testClasses = {
+            DefaultGroupsEndpoint.class,
+            TokenUtils.class
+    };
+    /**
+     * The test generated JWT token string
+     */
+    private String token;
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(testClasses)
+                    .addAsResource("publicKey.pem")
+                    .addAsResource("privateKey.pem")
+                    .addAsResource("applicationDefaultGroupsSignEncrypt.properties", "application.properties"));
+
+    @BeforeEach
+    public void generateToken() throws Exception {
+        token = Jwt.issuer("https://server.example.com").innerSign()
+                .keyAlgorithm(KeyEncryptionAlgorithm.RSA_OAEP)
+                .encrypt();
+    }
+
+    /**
+     * Validate a request with MP-JWT without a 'groups' claim is successful
+     * due to the default value being provided in the configuration
+     *
+     */
+    @Test
+    public void echoGroups() {
+        io.restassured.response.Response response = RestAssured.given().auth()
+                .oauth2(token)
+                .get("/endp/echo").andReturn();
+
+        Assertions.assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+        String replyString = response.body().asString();
+        // The missing 'groups' claim's default value, 'User' is expected
+        Assertions.assertEquals("User", replyString);
```

Comment:
You can use the RestAssured API to perform the assertions: 

```java 
given().auth()
                .oauth2(token)
                .get("/endp/echo")
.then().statusCode(200).
            body(".", equalTo("User"));
```

Reply:
@gastaldi Sure, it is the copy/paste, I'll update

Reply:
Hey @gastaldi - I also ended up cleaning up other tests - many of them have been copied across different projects - some of them require a deeper clean up - but I'll do it the next time I touch some of them...